### ISSUE TITLE
Amortized and Net Amortized Costs in UI

### ISSUE THREAD

**Issue by DKeeling01** [kubecost/cost-analyzer-helm-chart#1869]
**What problem are you trying to solve?**
Within the CUR there are several cost columns. Within the Kubecost UI, there aren't any columns for amortized and net amortized costs

**Describe the solution you'd like**
It would be very helpful for customers to be able to view the amortized and net amortized costs in the Kubecost UI
Formula for net amortized costs:
lineItem/NetUnblendedCost + savingsPlan/NetSavingsPlanEffectiveCost + reservation/NetEffectiveCost - reservation/NetRecurringFeeForUsage

**Describe alternatives you've considered**
NA - specific request from customer

**How would users interact with this feature?**
Customers would view and run reports via the Kubecost UI and API

**Comment by Adam-Stack-PM** [comment]
@DKeeling01, thanks for assigning me! cool idea.  Re routing to @dwbrown2 for product input and prioritization.

**Comment by AjayTripathy** [comment]
We're reworking some of the APIs here for cloud usage and expect to release broadly in v1.100!

**Comment by liorfranko** [comment]
+1

**Comment by teevans** [comment]
Closing - v1.100 should have this knocked out!

**Comment by AjayTripathy** [comment]
Awesome! Any documentation we can share with users to test this out @teevans and @Sean-Holcomb  ?

**Comment by kwombach12** [comment]
@teevans following up on Ajay's documentation request... how do I test this feature in the UI?

**Comment by teevans** [comment]
Going to have to rely on @Sean-Holcomb here - Anything you can share?

**Comment by Sean-Holcomb** [comment]
@kwombach12 The current UX is that netCost if available shows up in the Cost Column of the Cloud Cost page, and the difference between the netCost and cost is displayed in the credit column. Given that we do not have a any of the `net_` columns in our CUR, the change wont show up in our UI

**Comment by kwombach12** [comment]
@Sean-Holcomb I don't think I understand how the UX you're describing relates to cost amortization. Can you please clarify? I'm also happy to have a quick meeting with you if you'd rather walk me through it that way.

**Comment by Sean-Holcomb** [comment]
We currently only deal in amortized costs. Feature requests is more about net vs non-net

**Comment by teevans** [comment]
@Sean-Holcomb is there a good doc to understand the difference between the types of costs here? I think we're having a hard time understanding the actual definitions of what each type is

**Comment by kwombach12** [comment]
> We currently only deal in amortized costs. Feature requests is more about net vs non-net

So if I spin up a massive EC2 instance for one day that costs (making it up) 500k, that 500k doesn't appear on the day I spun it up? Rather it appears spread across the time range selected (7 days by default)?

**Comment by Sean-Holcomb** [comment]
Sorry @kwombach12  amortized in this context is referring to the cost of RI's and Savings Plan Nodes. Where their upfront cost is amortized across their term.

@teevans these are the cost explorer definitions
https://docs.aws.amazon.com/cost-management/latest/userguide/ce-exploring-data.html

The CUR definitions are here
https://docs.aws.amazon.com/cur/latest/userguide/Lineitem-columns.html
https://docs.aws.amazon.com/cur/latest/userguide/reservation-columns.html
https://docs.aws.amazon.com/cur/latest/userguide/savingsplans-columns.html

To try and get an equivalent of amortized cost from the cur we do
```
SUM(CASE line_item_line_item_type 
    WHEN 'DiscountedUsage' THEN reservation_effective_cost 
    WHEN 'SavingsPlanCoveredUsage' THEN savings_plan_savings_plan_effective_cost 
    ELSE line_item_unblended_cost END
) as amortized_cost
```
For Net Amortized Cost it is 
```
SUM(CASE line_item_line_item_type 
    WHEN 'DiscountedUsage' THEN reservation_net_effective_cost 
    WHEN 'SavingsPlanCoveredUsage' THEN savings_plan_net_savings_plan_effective_cost 
    ELSE line_item_net_unblended_cost END
) as amortized_cost
```

**Comment by udidn** [comment]
> @kwombach12 The current UX is that netCost if available shows up in the Cost Column of the Cloud Cost page, and the difference between the netCost and cost is displayed in the credit column. Given that we do not have a any of the `net_` columns in our CUR, the change wont show up in our UI

I understand that the difference between the net cost and the cost will be displayed in the "Credit" column in the Cloud Costs page.
However, will it be shown somewhere in the Allocations report page?
I think the real value of showing net amortized cost is that we can see it reflected in the in-cluster costs breakdown (that is, the Allocations report).
For example, if we aggregate the Allocations report by namespace, the expectation is one the the below (in my opinion, happy to hear other thoughts):

1. Each cost column (CPU, RAM, ... Total) will show the net amortized cost, but then since some customers prefer seeing the amortized costs (so, before discounts), Kubecost should give the option to change the report cost metric from amortized to net amortized
2. Show the costs as amortized (as I understand it is now), but somewhere in the Allocations report, show the difference between the amortized and net amortized per cost metric (CPU, RAM ... Total)

I just think that if Kubecost shows the net amortized only in the Cloud Costs page, it's missing the main purpose.
Cloud Costs can easily be seen in the cloud provider's tools (for example, Cost Explorer or CUR in AWS), where you can see the difference between amortized and net amortized costs.
The real value here is showing the net amortized costs for the in-cluster costs (the Allocations report).

So, to summarize - are the net amortized costs reflected in the Allocations report in the new version?
If not, is there a plan to do so?

**Comment by ivankube** [comment]
The net amortized cost will be shown on the Allocations page if users are using cloud cost in the version that is coming out today

Each cost-related column will show net amortized costs except for shared costs.

**Comment by udidn** [comment]
Hey team.

About a month ago, I opened [this ticket](https://kubecost.slack.com/archives/CLFV60Y90/p1680543405432679) to ask about the availability of net amortized costs in in the Allocations page.
The ticket was [concluded](https://kubecost.slack.com/archives/CLFV60Y90/p1680808655687509?thread_ts=1680543405.432679&cid=CLFV60Y90) stating that net amortized costs will be available in the Allocations page staring with version 1.103.
I see that [version 1.103.0 has been released](https://github.com/kubecost/cost-analyzer-helm-chart/releases/tag/v1.103.0) last week, but in the release notes, I don’t see any mention of the net amortized costs being available in the Allocations page.
Can someone please clarify if this feature is eventually in 1.103.0 or not?

Thanks

**Comment by AjayTripathy** [comment]
![Screen Shot 2023-05-08 at 2 39 31 PM](https://user-images.githubusercontent.com/453512/236941037-9a5fc818-3c85-497f-a390-3eb4bdd01296.png)

Hi Udi, this feature is available. You will need to enable the new cloudCost API like this:

```
kubecostModel:
  cloudCost:
    enabled: true
```

 And you should see a selector to flip between different pricing models on the /cloud page.

**Comment by udidn** [comment]
Really awesome, thanks!

**Comment by udidn** [comment]
But this screenshot is from the Cloud Costs page. My question is regarding the Allocations page.
Is the net amortized cost available in the Allocations page?

**Comment by Sean-Holcomb** [comment]
Once you enable CloudCost you can perform an reconciliation re run which will propagate those numbers to the assets page then the allocation page. `/etl/asset/reconciliation/run?commit=true`. Future allocations will be reconciled with net amortized.

**Comment by udidn** [comment]
Got it. Can I set a window for the `/etl/asset/reconciliation/run?commit=true` API?
If not, what's the window that is being used for the reconciliation re-run?

**Comment by udidn** [comment]
I found the answer [here](https://docs.kubecost.com/install-and-configure/install/cloud-integration#reconciliation-1). Thanks
